core/src/traits.rs

```diff
@@ -54,6 +56,7 @@ pub trait Client {
 }
 
 /// [JSON-RPC](https://www.jsonrpc.org/specification) client interface that can make requests, notifications and subscriptions.
+#[cfg(feature = "client")]
```

Comment:
I don't mind but then the `Client trait` should be feature gated too.

Also, I think it's fine to not feature gate these traits.